python/cudnn/sdpa/bwd/kernels/bprop_f16_sm80.py

```
@@ -0,0 +1,2038 @@
+    warps_per_sg: int = _LLAMA_CFG.WARPS_PER_SG,
+):
+    """Full SDPA backward — only ``O`` and ``lse`` (forward outputs) + ``dO`` are
+    needed beyond Q/K/V.  ``do_dot`` is computed on-device from O·dO unless
+    supplied.  ``mask`` / ``swa_window`` / ``right_bound`` / ``causal_bottom_right``
+    / ``seq_kv_lens`` mirror the forward kernel's masking (P=0 on masked cells →
+    dQ/dK/dV inherit it).  Returns (dQ, dK, dV) as io_dtype BSHD tensors."""
+    assert Q.dtype in (torch.float16, torch.bfloat16)
+    assert K.dtype == Q.dtype and V.dtype == Q.dtype and dO.dtype == Q.dtype and O.dtype == Q.dtype
+    io_is_bf16 = Q.dtype == torch.bfloat16
+    B, SQ, H, D = Q.shape
+    _, SKV, Hk, D_K = K.shape
```

Ruff (0.16.1)

[warning] Unpacked variable `D_K` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)